opensearch/variables.tf

```diff
@@ -6,22 +6,42 @@ variable "opensearch_host" {
     bootstrap_security  = bool
     host_name           = string
     initial_cluster     = bool
-    manager             = bool
+    cluster_manager     = optional(bool)
+    manager             = optional(bool)
   })
 }
 
 variable "opensearch_cluster" {
   description = "Opensearch cluster wide configurations"
   type = object({
-    auth_dn_fields      = object({
+    auth_dn_fields = object({
       admin_common_name = string
       node_common_name  = string
       organization      = string
     })
-    basic_auth_enabled  = bool
-    cluster_name        = string
-    seed_hosts          = list(string)
-    verify_domains      = bool
+    basic_auth_enabled            = bool
+    cluster_name                  = string
+    seed_hosts                    = list(string)
+    initial_cluster_manager_nodes = optional(list(string))
+    verify_domains                = bool
+
+    audit = optional(object({
+      enabled = optional(bool, false)
+      index   = string
+
+      external = optional(object({
+        http_endpoints = list(string)
+        auth = optional(object({
+          ca_cert     = optional(string, "")
+          client_cert = optional(string, "")
+          client_key  = optional(string, "")
+          username    = optional(string, "")
+          password    = optional(string, "")
+        }), null)
+      }), null)
```

Comment:
I would avoid the nulls as default values as I see it adds non-negligible complexity in the rest of your code. 

I would either not set default values (a pattern we use in the rest of our cloudinit orchestration) or set the structures with empty string and false boolean values as defaults. It will make the rest of your code prettier.